Lib/test/_test_multiprocessing.py

```diff
@@ -1348,23 +1348,32 @@ class _TestLock(BaseTestCase):
     def test_lock(self):
         lock = self.Lock()
         self.assertEqual(lock.acquire(), True)
+        self.assertEqual(lock.locked(), True)
         self.assertEqual(lock.acquire(False), False)
         self.assertEqual(lock.release(), None)
+        self.assertFalse(lock.locked())
         self.assertRaises((ValueError, threading.ThreadError), lock.release)
 
     def test_rlock(self):
         lock = self.RLock()
         self.assertEqual(lock.acquire(), True)
+        self.assertTrue(lock.locked())
         self.assertEqual(lock.acquire(), True)
         self.assertEqual(lock.acquire(), True)
         self.assertEqual(lock.release(), None)
+        self.assertTrue(lock.locked())
         self.assertEqual(lock.release(), None)
         self.assertEqual(lock.release(), None)
+        self.assertFalse(lock.locked())
         self.assertRaises((AssertionError, RuntimeError), lock.release)
 
     def test_lock_context(self):
-        with self.Lock():
-            pass
+        with self.Lock() as locked:
+            self.assertTrue(locked)
```

Comment:
Maybe you wanted to mean `lock.locked()` instead? or is the test only meant for testing the `with`?

Reply:
It is testing `with` 🤔 
Not sure that I understand what you mean.

Reply:
I wondered why the `pass` statement was not sufficient but I guess it's just to ensure that we "do" something inside the with body?

Reply:
because we actually want to test what `__enter__` returns, it is not ever tested. It returns a bool.

Reply:
Ah! I never knew it returned a bool actually!